File: Loenn/entities/strawberry_jam/expiring_dash_refill.lua
local expiringDashRefill = {}

expiringDashRefill.name = "CommunalHelper/SJ/ExpiringDashRefill"

expiringDashRefill.fieldInformation = {
    dashExpirationTime = {
        fieldType = "number",
        minimumValue = 0.0
    },
    hairFlashThreshold = {
        fieldType = "number",
        minimumValue = 0.0,
        maximumValue = 1.0
    }
}

expiringDashRefill.placements = {
    {
        name = "expiring_dash_refill",
        data = {
            oneUse = false,
            dashExpirationTime = 5.0,
            hairFlashThreshold = 0.2
        }
    }
}

expiringDashRefill.texture = "objects/refill/idle00"

return expiringDashRefill

File: Ahorn/entities/strawberryJam/ExpiringDashRefill.jl
module CommunalHelperSJExpiringDashRefill

using ..Ahorn, Maple

@mapdef Entity "CommunalHelper/SJ/ExpiringDashRefill" ExpiringDashRefill(
    x::Integer,
    y::Integer,
    oneUse::Bool=false,
    dashExpirationTime::Number=5.0,
    hairFlashThreshold::Number=0.2
)

const placements = Ahorn.PlacementDict(
    "Expiring Dash Refill (Strawberry Jam) (Communal Helper)" => Ahorn.EntityPlacement(
        ExpiringDashRefill
    )
)

const spriteOneDash = "objects/refill/idle00"

function Ahorn.selection(entity::ExpiringDashRefill)
    x, y = Ahorn.position(entity)

    return Ahorn.getSpriteRectangle(spriteOneDash, x, y)
end

Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::ExpiringDashRefill) = Ahorn.drawSprite(ctx, spriteOneDash, 0, 0)

end